console: Add Azure & Keycloak docs (#1429)
redpanda-data/documentation: docs/manage/security/console/authentication.mdx

---

<head>
  <meta
    name="title"
    content="SSO Authentication in Redpanda Console | Redpanda Docs"
  />
  <meta
    name="description"

>     content="Configure authentication with external identity providers such as Google, GitHub or Okta in Redpanda Console."

Author: @AnneLeeperRP Should we update this as there are more providers now?

Reviewer: Good point. Here are some questions to think about:
* How many IdPs will we support in the future? (Too many to list?)
* If we only mention some of them in this description (and we say "such as", which implies the list is not exhaustive), which ones deserve to be mentioned? The most commonly used ones?
* What about leaving out specify IdPs and just saying "Configure authentication with external identity providers in Redpanda Console"?

I think I like the last idea the best.

Author: > How many IdPs will we support in the future? (Too many to list?)

Possibly we are already there.

> If we only mention some of them in this description (and we say "such as", which implies the list is not exhaustive), which ones deserve to be mentioned? The most commonly used ones?

Possibly. I think the listed three are the most commonly used ones. Maybe also AzureAD

> What about leaving out specify IdPs and just saying "Configure authentication with external identity providers in Redpanda Console"?

Also okay for me

  />
</head>

:::info
This feature requires an [Enterprise license](../../../../get-started/licenses). To upgrade, contact [Redpanda sales](https://redpanda.com/try-redpanda?section=enterprise-cloud).
:::

Redpanda Console supports authentication via OAuth 2.0 or OIDC for external identity providers, such as:

- [AzureAD](../azure-ad)
- [Google](../google)
- [GitHub](../github)
- [Keycloak](../keycloak)
- [Okta](../okta)
- [Generic OIDC](../generic-oidc)

> You can use one or more login providers at the same time. To enable SSO authentication

Reviewer: Add comma after introductory phrase:
```suggestion
You can use one or more login providers at the same time. To enable SSO authentication,
```

login:
  enabled: true

  # jwtSecret is the secret key you must use to sign and encrypt the JSON
  # web token used to store user sessions. This secret key is
  # critical for the security of Redpanda Console's authentication and
  # authorization system. Use a long, complex key with a combination of

>   # numbers, letters, and special characters. While you must use a minimum of
>   # 10 characters, Redpanda recommends using more than 32
>   # characters. For additional security, use a different secret key for
>   # each environment. jwtSecret can be securely generated with the following

Reviewer: ```suggestion
  # numbers, letters, and special characters. The minimum number of
  # characters is 10, but Redpanda recommends using more than 32
  # characters. For additional security, use a different secret key for
  # each environment.
  
  jwtSecret can be securely generated with the following
```

Author: This comment is copy pasted for consistency across all IdPs. This had been reviewed for other pages before, thus I prefer to keep it as is or change everything if needed.

  # command: LC_ALL=C tr -dc '[:alnum:]' < /dev/random | head -c32
  #

>   # If you update this secret key, any users who are
>   # already logged into Redpanda Console will be logged out and will have

Reviewer: "logged into" should be "logged in to" (and I suggested changing "update" to "change"):

```suggestion
  # If you change this secret key, any users who are
  # already logged in to Redpanda Console will be logged out and will have
```

Author: This comment is copy pasted for consistency across all IdPs. This had been reviewed for other pages before, thus I prefer to keep it as is or change everything if needed.
